### PR TITLE
[JBIDE-24876] fixed wait for dc

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by adietish** [comment]
@bdshadow @jeffmaury please review and merge once https://github.com/jbosstools/jbosstools-openshift/pull/1544 was merged in

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftLaunchController.java [hunk 1]
```diff
@@ -120,29 +124,40 @@ private String getLabel(String mode) {
...
+		while (!metadata.load()) {
+			if (!sleep(RECHECK_DELAY, elapsed, WAIT_FOR_DOCKERIMAGELABELS_TIMEOUT, monitor)) {
+				return false;
+			};
 		}
-		return false;
+		return true;
+	}
+
+	private boolean sleep(int sleep, int elapsed, int timeout, IProgressMonitor monitor) {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftLaunchController.java]: this is clearly a false positive. The method is being used.

**Comment by adietish** [comment]
testPR

**Comment by robstryker** [comment]
I suggested some changes regarding timeout to andre. +1 once he fixes that.

**Comment by adietish** [comment]
testPR

**Pull request merged**